test/compare_results.py

```diff
@@ -3,65 +3,118 @@
+    "einstein": 2,
+    "nu": 3,
+}
 
 def find_start_end_block(lines, blockname):
-    """Identify start and end of blocks"""
+    """Identify start and end of blocks in .chk files"""
     for i, line in enumerate(lines):
         if "Start " + blockname in line:
             start_idx = i
         elif "End " + blockname in line:
             end_idx = i
     return start_idx, end_idx
 
-def extract_quantum_block(lines):
-    """Extract only the quantum block from a chk file"""
-    idxs = find_start_end_block(lines, "Quantum")
-    temp = lines[idxs[0]+4:idxs[-1]]
-    return [line.split() for line in temp]
+def find_log_block(lines, blockname):
+    """Identify start and end of block in log file"""
+    start_idx = end_idx = None
+    for i, line in enumerate(lines):
+        if blockname in line:
+            start_idx = i
+            break
+
+     # find first "done" after we've found blockname
+    for i, line in enumerate(lines[start_idx:]):
+        if "done" in line:
+            end_idx = i + start_idx
+            break
 
-def extract_quantum_energies(block):
-    """Extract quantum energies from entire block"""
-    return [float(line[QUANTUM_ENERGY_IDX]) for line in block]
+    if start_idx is not None:
+        raise Exception(f"{blockname} not found")
+    elif end_idx is not None:
+        raise Exception(f"Could not find end of {blockname}")
+
+    return start_idx, end_idx
 
 def read_chk_file(fname):
     """Read checkpoint file as a list of lines"""
     with open(fname, 'r') as fp:
         # Strip newlines and lines with comments
-        lines = [line for line in fp.readlines() if line != '\n' and '<-' not in line]
+        lines = fp.readlines()
     return lines
 
+def strip_newlines(lines):
+    """Remove all lines which are just newlines"""
+    return [line for line in lines if line != '\n']
+
+def strip_comments(lines):
+    """Remove all lines with comments"""
+    return [line for line in lines if '<-' not in line]
+
+def extract_column(lines, column_no):
+    """Extract column of numbers from list of str lines"""
+    return [float(line.split()[column_no]) for line in lines]
+
 def read_energy_column(fname, column_no):
     """Extract energies from a column in file fname"""
     lines = read_chk_file(fname)
-    lines = lines[:-1] # remove last line (which is not part of the actual data)
-    return [float(line.split()[column_no]) for line in lines]
+    lines = strip_newlines(strip_comments(lines))
+    # remove last line (which is not part of the actual data)
+    lines = lines[:-1]
+    return extract_column(lines, column_no)
 
-def read_quantum_block(fname):
+def read_quantum_energies(fname):
     """Extract quantum energies from fname"""
     lines = read_chk_file(fname)
-    return extract_quantum_block(lines)
+    lines = strip_newlines(strip_comments(lines))
+    start, end = find_start_end_block(lines, "Quantum")
+    # take out first 4 lines and last line of block
+    lines = lines[start+4:end]
```

Comment:
I think this way is clearer than the previous one with `idxs[0]` and `idxs[1]`,  good choice.